### PR TITLE
docs(bazel): add aio/guide/bazel

### PR DESCRIPTION
**Pull request by alexeagle**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by jenniferfell** [comment]
See https://github.com/jenniferfell/angular branch jfell-bazel. 

There are 4 commits, with separate suggestions. Feel free to take one or all.  
* Add bazel doc to nav
* Edit bazel doc
* Add link from CLI overview to bazel doc
* In-page nav recommendations. fwiw it's always implied in doc that basic use cases come first and advanced later, so i don't think the last ones need to be L3. Having them grouped made it feel to me like they all are tied to using bazel directly, but technically i don't think that was the intent. I think they are distinct un-teathered use cases. Alternate idea: Keep Advanced Usage heading. Make everything else H3 and create a heading for the first part about configuration, so it's clear that under Advanced Usage there is cli-related config, running directly, querying the graph, and customizing bazel build files.

**Review by jenniferfell — APPROVED** [review]
LGTM from a doc experience and editorial POV. I'm not the "please squash your commits" police. :-)

**Comment by alexeagle** [comment]
caretaker:global approved

File: aio/content/guide/bazel.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Building with Bazel
+
+This guide explains how to build and test Angular apps with Bazel. 
+
+
+<div class="alert is-helpful">
+
+
+This guide assumes you are already familiar with developing and building Angular applications using the [CLI](cli). 
+It also assumes that you are familiar with [Bazel](https://docs.bazel.build).
```
Comment (IgorMinar) [inline aio/content/guide/bazel.md]: is this true? most CLI users don't need to be familiar with Bazel unless they want to do more advanced stuff. could we clarify that only advanced use-cases will require Bazel knowledge.

File: aio/content/guide/bazel.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+
+
+## Running Bazel directly
+
+In some cases you'll want to bypass the Angular CLI builder, and run the Bazel CLI directly.
+
+The common commands in Bazel are:
+
+* `bazel build [targets]`: Compile the default output artifacts of the given targets.
```
Comment (IgorMinar) [inline aio/content/guide/bazel.md]: these commands will not work unless you install @angular/bazel globally which is an optional step mentioned above. Maybe add a warning in this section that in order to run bazel directly, one must install @angular/bazel globally.

File: aio/content/guide/bazel.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+Using the graphviz optional dependency, you'll have a program `dot`, which you can use with `bazel query`:
+
+```bash
+$ bazel query --output=graph ... | dot -Tpng > graph.png
+```
+
+See https://docs.bazel.build/versions/master/query-how-to.html for more details on `bazel query`.
+
+
+## Customizing `BUILD.bazel` files
```
Comment (IgorMinar) [inline aio/content/guide/bazel.md]: add a warning about risks of customizing the build files - especially api stability and our inability to optimize the default build configuration for you... so you essentially `ng eject` but are working directly with apis that are still not stable.

**Review by IgorMinar — APPROVED** [review]
lgtm, but I'd add a stronger language about this being labs and experimental.

File: aio/content/guide/bazel.md (hunk 1 above)
Comment (alexeagle) [reply aio/content/guide/bazel.md]: moved below to advanced section

File: aio/content/guide/bazel.md (hunk 2 above)
Comment (alexeagle) [reply aio/content/guide/bazel.md]: added

File: aio/content/guide/bazel.md (hunk 3 above)
Comment (alexeagle) [reply aio/content/guide/bazel.md]: added

**Pull request closed without merging**